File: girl/src/main/java/com/ayvytr/girl/view/fragment/GirlsFragment.java
package com.ayvytr.girl.view.fragment;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.StaggeredGridLayoutManager;
import android.view.View;

import com.alibaba.android.arouter.launcher.ARouter;
import com.ayvytr.baseadapter.MultiItemTypeAdapter;
import com.ayvytr.commonlibrary.bean.BaseGank;
import com.ayvytr.commonlibrary.bean.Gank;
import com.ayvytr.commonlibrary.constant.GirlsConstant;
import com.ayvytr.girl.R;
import com.ayvytr.girl.adapter.GirlsAdapter;
import com.ayvytr.girl.contract.GirlsContract;
import com.ayvytr.girl.presenter.GirlsPresenter;
import com.ayvytr.mvpbaselist.BaseListFragment;
import com.scwang.smartrefresh.layout.api.RefreshLayout;

import java.util.ArrayList;

import butterknife.ButterKnife;

/**
 * @author admin
 */
public class GirlsFragment extends BaseListFragment<GirlsPresenter, Gank> implements GirlsContract.View {
    @Override
    protected GirlsPresenter getPresenter() {
        return new GirlsPresenter(this);
    }

    @Override
    public void initExtra() {

    }

    @Override
    public void initView(@Nullable Bundle savedInstanceState) {
        super.initView(savedInstanceState);
        ButterKnife.bind(this, mContentView);
        mAdapter = new GirlsAdapter(getContext(), R.layout.layout_item_girl);
        mRvList.setLayoutManager(new StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL));
        mAdapter.setOnItemClickListener(new MultiItemTypeAdapter.OnItemClickListener() {
            @Override
            public void onItemClick(View view, RecyclerView.ViewHolder holder, int position) {
                ArrayList list = new ArrayList<Gank>();
                list.addAll(mAdapter.getDatas());

                ARouter.getInstance()
                       .build(GirlsConstant.PHOTO_VIEW)
                       .withParcelableArrayList(GirlsConstant.EXTRA_PHOTOS, list)
                       .withInt(GirlsConstant.EXTRA_PHOTO_POSITION, position)
                       .navigation();
            }
        });
        mRvList.setAdapter(mAdapter);
    }

    @Override
    public void initData(@Nullable Bundle savedInstanceState) {
        mPresenter.requestGankMm(pageSize, currentPage);
    }

    @Override
    public int getContentViewRes() {
        return R.layout.layout_list;
    }

    @Override
    public void onLoadMore(@NonNull RefreshLayout refreshLayout) {
        super.onLoadMore(refreshLayout);
        mPresenter.requestGankMm(pageSize, currentPage);
    }

    @Override
    public void onRefresh(@NonNull RefreshLayout refreshLayout) {
        super.onRefresh(refreshLayout);
        mPresenter.requestGankMm(pageSize, currentPage);
    }

    @Override
    public void showGankMm(BaseGank gank) {
        updateList(gank.getResults());
    }
}

File: girl/src/main/java/com/ayvytr/girl/view/fragment/GirlsFragment.kt
package com.ayvytr.girl.view.fragment

import android.os.Bundle
import android.support.v4.app.ActivityOptionsCompat
import android.support.v7.widget.StaggeredGridLayoutManager
import android.view.ViewGroup
import com.alibaba.android.arouter.launcher.ARouter
import com.ayvytr.baselist.BaseListFragment
import com.ayvytr.commonlibrary.bean.BaseGank
import com.ayvytr.commonlibrary.bean.Gank
import com.ayvytr.commonlibrary.constant.GirlsConstant
import com.ayvytr.girl.R
import com.ayvytr.girl.adapter.GirlsAdapter
import com.ayvytr.girl.adapter.callback.GirlsCallback
import com.ayvytr.girl.contract.GirlsContract
import com.ayvytr.girl.presenter.GirlsPresenter
import com.scwang.smartrefresh.layout.api.RefreshLayout
import java.util.*


/**
 * @author admin
 */
class GirlsFragment : BaseListFragment<GirlsPresenter, Gank>(), GirlsContract.View {
    override fun getPresenter(): GirlsPresenter {
        return GirlsPresenter(this)
    }

    override fun initExtra() {

    }

    override fun initView(savedInstanceState: Bundle?) {
        super.initView(savedInstanceState)
        mAdapter = GirlsAdapter(context!!, R.layout.layout_item_girl)
        mRvList.layoutManager = StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL)
        mAdapter.setOnItemClickListener { view, holder, position ->
            val list = ArrayList(mAdapter.datas)

            val compat = ActivityOptionsCompat.makeSceneTransitionAnimation(activity!!,
                                                                            view, getString(R.string.trans_photo_name))
            ARouter.getInstance()
                .build(GirlsConstant.PHOTO_VIEW)
                .withParcelableArrayList(GirlsConstant.EXTRA_PHOTOS, list)
                .withInt(GirlsConstant.EXTRA_PHOTO_POSITION, position)
                .withOptionsCompat(compat)
                .navigation(context)
        }
        mRvList.adapter = mAdapter
    }

    override fun initData(savedInstanceState: Bundle?) {
        mPresenter.requestGankMm(mPageSize, mCurrentPage)
    }

    override fun getContentViewRes(): Int {
        return R.layout.layout_list
    }

    override fun onLoadMore(refreshLayout: RefreshLayout) {
        super.onLoadMore(refreshLayout)
        mPresenter.requestGankMm(mPageSize, mCurrentPage + 1)
    }

    override fun onRefresh(refreshLayout: RefreshLayout) {
        super.onRefresh(refreshLayout)
        mPresenter.requestGankMm(mPageSize, 1)
    }

    override fun showGankMm(gank: BaseGank, currentPage: Int) {
        mCurrentPage = currentPage
        updateList(gank.results, GirlsCallback(mAdapter.datas, gank.results!!))
    }

    override fun onDestroyView() {
        super.onDestroyView()
        //内存泄漏
        if(mContentView != null){
            (mContentView.parent as ViewGroup).removeView(mContentView)
        }
    }
}
